st2common/st2common/services/workflows.py

```diff
@@ -1129,3 +1130,55 @@ def update_execution_records(wf_ex_db, conductor, update_lv_ac_on_statuses=None,
     if status_changed and wf_lv_ac_db.status in ac_const.LIVEACTION_COMPLETED_STATES:
         LOG.info('[%s] Workflow action execution is completed and invoking post run.', wf_ac_ex_id)
         runners_utils.invoke_post_run(wf_lv_ac_db)
+
+
+@retrying.retry(retry_on_exception=wf_exc.retry_on_exceptions)
+def request_rerun(ac_ex_db, st2_ctx, options):
+    ac_ex_id = str(ac_ex_db.id)
+    LOG.info('[%s] Processing rerun request for workflow.', ac_ex_id)
+
+    wf_ex_id = st2_ctx['workflow_execution_id']
+    try:
+        wf_ex_db = wf_db_access.WorkflowExecution.get_by_id(wf_ex_id)
+    except db_exc.StackStormDBObjectNotFoundError:
+        msg = '[{0}] Unable to find workflow execution db with id "{1}"'.format(ac_ex_id, wf_ex_id)
+        raise wf_exc.WorkflowExecutionRerunException(msg)
+
+    if wf_ex_db.status != statuses.FAILED:
+        msg = '[{0}] Workflow execution "{1}" is not rerunable because its status is not "failed"'.\
+            format(ac_ex_id, wf_ex_id)
+        raise wf_exc.WorkflowExecutionRerunException(msg)
```

Comment:
For consistency, use `%s` instead of format to substitute text.  So, please change to the following code snippet.  Also, I separated the message and the substitution to keep within 100 line length and w/o having to do separate lines and weird indentation.  This is consistent with how I coded the workflow service.

```
msg = '[%s] Workflow execution "%s" is not rerunable because its not in a failed state.'
raise wf_exc.WorkflowExecutionRerunException(msg % (ac_ex_id, wf_ex_id))
```